server/command_handler.go

```diff
@@ -1,15 +1,21 @@
 package server
 
 import (
+	"bytes"
 	"github.com/MohamedShetewi/stomp-in-go/frame"
 	"github.com/MohamedShetewi/stomp-in-go/utils"
 )
 
 type commandHandler func(server *Server, conn *Client, frame *frame.Frame)
 
 var commandHandlerMap = map[frame.Command]commandHandler{
-	frame.CONNECT:   connectHandler,
-	frame.SUBSCRIBE: subscribeHandler,
+	frame.CONNECT:     connectHandler,
+	frame.SUBSCRIBE:   subscribeHandler,
+	frame.UNSUBSCRIBE: unsubscribeHandler,
+	frame.SEND:        sendHandler,
+	frame.BEGIN:       beginHandler,
+	frame.COMMIT:      commitHandler,
+	frame.ABORT:       abortHandler,
 }
 
 func connectHandler(server *Server, client *Client, frm *frame.Frame) {
```

Comment:
DESIGN;
If I understand correctly, you are trying to decouple the way you handle the commands from the server and the clients by using dependency injection; decoupling is good. but here you are still coupled as you only accept pointers to a specific structs (Server, Client). It would be better to use interfaces in this case, you will need to think how these entities interact with each other and write interfaces for that interaction.